libsplinter/src/service/scabbard/rest_api.rs

```diff
@@ -272,27 +269,23 @@ pub fn make_get_batch_status_endpoint() -> ServiceEndpoint {
                 })
                 .map(Duration::from_secs);
 
-            let timeout = match wait {
-                Some(wait_time) => match Instant::now().checked_add(wait_time) {
-                    Some(t) => Some(t),
-                    None => {
-                        error!("Failed to determine wait time");
-                        return Box::new(
-                            HttpResponse::InternalServerError()
-                                .json(json!({
-                                    "message": "An internal error occurred"
-                                }))
-                                .into_future(),
-                        );
-                    }
+            match scabbard.get_batch_info(ids, wait) {
+                Ok(iter) => match iter.collect::<Result<Vec<_>, _>>() {
+                    Ok(batch_infos) => Box::new(HttpResponse::Ok().json(batch_infos).into_future()),
+                    Err(err) => Box::new(
+                        HttpResponse::RequestTimeout()
```

Comment:
openapi spec should be updated with this new error response

Reply:
Added to the OpenAPI spec